docker/docker.go

```diff
@@ -109,3 +109,20 @@ func (service *InfoService) GetServiceNameFromDockerEngine(containerName string)
 
 	return containerInspect.Config.Labels[serviceNameLabel], nil
 }
+
+// IsLeaderNode returns true if the node is the leader node of the swarm cluster
+func (service *InfoService) IsLeaderNode() (bool, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion(agent.SupportedDockerAPIVersion))
+	if err != nil {
+		return false, err
+	}
+
+	defer cli.Close()
+
+	node, _, err := cli.NodeInspectWithRaw(context.Background(), "self")
```

Comment:
@deviantony would love your help here:
this fails on worker nodes, which should be ok because it fails also when using the cli. Maybe in this case we should just return false?
This also fails on manager nodes, because it doesn't recognize node `self`, but in the cli `docker node inspect self` works.

Maybe I should do it differently? list all nodes and find the leader?